pkg/cmd/container/remove.go

```diff
@@ -0,0 +1,218 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package container
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"syscall"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/cio"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/containerd/nerdctl/pkg/api/types"
+	"github.com/containerd/nerdctl/pkg/clientutil"
+	"github.com/containerd/nerdctl/pkg/cmd/volume"
+	"github.com/containerd/nerdctl/pkg/dnsutil/hostsstore"
+	"github.com/containerd/nerdctl/pkg/idutil/containerwalker"
+	"github.com/containerd/nerdctl/pkg/labels"
+	"github.com/containerd/nerdctl/pkg/namestore"
+	"github.com/sirupsen/logrus"
+)
+
+var _ error = ErrContainerStatus{}
+
+// ErrContainerStatus represents an error that container is in a status unexpected
+// by the caller. E.g., remove a non-stoped/non-created container without force.
+type ErrContainerStatus struct {
+	ID     string
+	Status containerd.ProcessStatus
+}
+
+func (e ErrContainerStatus) Error() string {
+	return fmt.Sprintf("container %s is in %v status", e.ID, e.Status)
+}
+
+// NewStatusError creates an ErrContainerStatus from container id and status.
+func NewStatusError(id string, status containerd.ProcessStatus) error {
+	return ErrContainerStatus{
+		ID:     id,
+		Status: status,
+	}
+}
```

Comment:
This part is new that replaces the old `statusError` since after moving to `pkg`, other `cmd` cannot access the `error` within it.